include/mbedtls/aes.h

```diff
@@ -329,6 +329,57 @@ int mbedtls_aes_crypt_ctr( mbedtls_aes_context *ctx,
                        unsigned char *output );
 #endif /* MBEDTLS_CIPHER_MODE_CTR */
 
+#if defined(MBEDTLS_CIPHER_MODE_XTS)
+/**
+ * \brief      This function performs an AES-XTS encryption or decryption
+ *             operation for an entire XTS sector.
+ *
+ *             The key for tweak computation is separate from the key for AES
+ *             block encryption. So, two AES contexts must be supplied. XXX
+ *             Design review: It's a bit strange to have two AES ctxs.
```

Comment:
Unfortunately the extra key does not fit in the existing context in all cases so we can't store the second key there. Adding a pointer would be a dead weight in all the other cases, so I can see why you didn't go with it. I feel that adding a new context adds complexity and it is something that I too would be reluctant to do.

However I think that there is less opportunity to misuse the API if there is a dedicated XTS context and in this instance I would vote for defensive coding over conceptual cleanliness.

I was thinking something like this:
``` C
typedef struct
{
    mbedtls_aes_context sub_ctx[2];
}
mbedtls_aes_xts_context;
```

Reply:
That is even better, good idea!

Reply:
@yanesca Would you also recommend new `mbedtls_aes_xts_setkey_enc(mbedtls_aes_xts_context* ctx)` and `mbedtls_aes_xts_setkey_dec(mbedtls_aes_xts_context* ctx)` functions?

Reply:
Yes, I think that would be the cleanest solution. Especially if we add the usual `_init()` and `_free()` along with them.

Reply:
I've changed the AES XTS adding commit to introduce a new context type and the associated functions.

Reply:
Thanks! The API looks good to me now.